core/services/ocr2/validate/validate.go

```diff
@@ -178,11 +184,58 @@ func validateOCR2GenericPluginSpec(jsonConfig job.JSONConfig) error {
 		return errors.New("generic config invalid: must provide plugin name")
 	}
 
-	if p.TelemetryType == "" {
-		return errors.New("generic config invalid: must provide telemetry type")
+	if p.OCRVersion != 2 && p.OCRVersion != 3 {
+		return errors.New("generic config invalid: only OCR version 2 and 3 are supported")
 	}
 
-	return nil
+	plugEnv := env.NewPlugin(p.PluginName)
+
+	command := p.Command
+	if command == "" {
+		command = plugEnv.Cmd.Get()
+	}
+
+	if command == "" {
+		return errors.New("generic config invalid: no command found")
+	}
+
+	_, err = exec.LookPath(command)
+	if err != nil {
+		return fmt.Errorf("failed to find binary  %q", command)
+	}
+
+	envVars, err := plugins.ParseEnvFile(plugEnv.Env.Get())
+	if err != nil {
+		return fmt.Errorf("failed to parse median env file: %w", err)
+	}
+	if len(p.EnvVars) > 0 {
+		for k, v := range p.EnvVars {
+			envVars = append(envVars, k+"="+v)
+		}
+	}
+
+	loopID := fmt.Sprintf("%s-%s-%s", p.PluginName, spec.ContractID, spec.GetID())
+	cmdFn, grpcOpts, err := rc.RegisterLOOP(plugins.CmdConfig{
```

Comment:
It seems a bit odd that we're registering the LOOPP only to immediately unregister it. Do we even need to register it, and if so, why?

Reply:
When the plugin is started it requires a prometheusPort, in order to get a valid port we must register the LOOPP

Reply:
hmm, this seems inverted.

what is the goal of this code? 

another red flag is starting and closing the plugin. why is that happening?

if we register and unregister for the purpose of getting a port and validating a port, then we are still open to failure later when re-register it because there is dynamic code to allocate the port. moreover, the port will be different, so what use is the first registration?

Reply:
@krehermann To validate the config we must start the binary so we can call the `validateConfig()` that is inside the LOOPP, this has to happen inside the LOOPP because we do not know what configuration it expects. When the LOOPP is started, as part of the registration, a Prometheus port is expected, that is why we are registering the LOOPP.

Reply:
@krehermann We thought about this quite a bit, and @george-dorin is correct -- I agree that it's counter-intuitive to register and deregister immediately but it's not clear what alternative there is given that the point of registering is to "lock" host resources such as the prometheus port.

Reply:
The point of this code is just to call out the LOOPP to validate the config; we aren't trying to do any validation with respect to allocation of system resources and we aren't guaranteeing that a successful validation implies that we will definitely start a LOOPP either.